### ISSUE TITLE
CreateDictionaryEntry behavior when `word` is empty string

### ISSUE THREAD

**Issue by mammothb** [wolfgarbe/SymSpell#118]
I noticed a strange behavior with [Edits](https://github.com/wolfgarbe/SymSpell/blob/master/SymSpell/SymSpell.cs#L739) method when `maxDictionaryEditDistance` is larger than some of the words being added the dictionary and would like to understand the rationale/reason behind it.

The following code snippet replicate the observation
```csharp
const int initialCapacity = 82834;
const int maxEditDistance = 5;
const int prefixLength = 7;
var symSpell = new SymSpell(initialCapacity, maxEditDistance, prefixLength);

symSpell.CreateDictionaryEntry("asdf", 2);
symSpell.CreateDictionaryEntry("qwer", 2);
symSpell.CreateDictionaryEntry("zxcv", 2);
symSpell.CreateDictionaryEntry("uiopjkl", 2);
symSpell.CreateDictionaryEntry("", 2);

foreach (KeyValuePair<int, string[]> entry in symSpell.deletes)
{
    string output = " (" + entry.Key + ") [";
    foreach (string val in entry.Value)
    {
        output += "\"" + val + "\", ";
    }
    output += "]";
    Console.WriteLine(output);
}
```
The list of deletes for the empty string `""` is `(18652612) ["asdf", "qwer", "zxcv", "", ]` which contains all the dictionary words that are below the `maxEditDistance`. I assume this is because at a max edit distance of 5, "asdf", "qwer", and "zxcv" which are 4 edit distance away from "" are viable candidates. 

However, may I know if there are cases where this entry will be actually selected and used to select a candidate? So far I can only think of doing `Lookup("", Verbosity.All)` but that will still return "" as the top result since it's an exact match but the other results are unused.

**Comment by wolfgarbe** [comment]
This has been a fix introduced in v5.0 https://github.com/wolfgarbe/SymSpell#changes-in-v50 
It ensures that suggestions for input_words with Length <= editDistanceMax are complete. 
e.g. input_word "axe" has a length <= maxEditDistance 5. One of the input_deletes becomes "".
And the we have a match with the dictionary_delete "". 
And the dictionary_delete "" points to all dictionary_words with length <= maxEditDistance as suggestions.

**Comment by mammothb** [comment]
Yes, I understand how it works better now, thanks for the explanation!
